pootle/apps/pootle_translationproject/models.py

```diff
@@ -43,51 +49,154 @@ def __init__(self, parent):
         self.termmatchermtime = None
 
 
-def create_or_resurrect_translation_project(language, project):
-    tp = create_translation_project(language, project)
-    if tp is not None:
-        if tp.directory.obsolete:
-            tp.directory.obsolete = False
-            tp.directory.save()
-            logging.info(u"Resurrected %s", tp)
-        else:
-            logging.info(u"Created %s", tp)
-
-
 def create_translation_project(language, project):
     from pootle_app import project_tree
     if project_tree.translation_project_should_exist(language, project):
         try:
-            translation_project, created = TranslationProject.objects.all() \
-                    .get_or_create(language=language, project=project)
-            return translation_project
+            tp, created = TranslationProject.objects \
+                                            .get_or_create(language=language,
+                                                           project=project)
+            if created:
+                logging.info(u"Created %s", tp)
+
+            return tp
         except OSError:
             return None
         except IndexError:
             return None
 
 
-def scan_translation_projects(languages=None, projects=None):
-    project_query = Project.objects.enabled()
+def update_translation_project(language, project, **options):
```

Comment:
The Python convention to name extra args is `**kwargs` — let's use it as we do elsewhere.